src/test/java/org/opentripplanner/middleware/cdp/ConnectedDataPlatformTest.java

```diff
@@ -0,0 +1,239 @@
+        // prevents a delete error under the tidy-up process.
+        tripRequest = null;
+        tripSummary = null;
+        fileContents = getContentsOfFileInZip(
+            tempFile,
+            getFileName(startOfYesterday, ConnectedDataManager.DATA_FILE_NAME_SUFFIX)
+        );
+        tripHistory = JsonUtils.getPOJOFromJSON(fileContents, TripHistory.class);
+        assertTrue(tripHistory.tripRequests.stream().noneMatch(tripRequest -> tripRequest.userId.equals(userId)));
+    }
+
+    /**
+     * If the system is down for a period of time, make sure that the days between the last upload and the current day
+     * are correctly staged.
+     */
+    @Test
+    public void canCorrectlyStageDays() {
+        Date sevenDaysAgo = getStartOfDay(getDateMinusNumberOfDays(new Date(), 7));
+        Set<LocalDate> betweenDays = DateTimeUtils.getDatesBetween(
+            getDatePlusNumberOfDays(sevenDaysAgo,1),
+            new Date()
+        );
+        TripHistoryUpload tripHistoryUpload = new TripHistoryUpload(sevenDaysAgo);
+        Persistence.tripHistoryUploads.create(tripHistoryUpload);
+        ConnectedDataManager.stageUploadDays();
+        assertEquals(
+            betweenDays.size(),
+            Persistence.tripHistoryUploads.getCountFiltered(Filters.gt("uploadDate", sevenDaysAgo))
+        );
+    }
+
+    /**
+     * Add an OTP user with 'storeTripHistory' set to true and related trip requests/summaries. Via the API update the
+     * 'storeTripHistory' to false and confirm that the trip history is removed and the appropriate days are flagged for
+     * updating.
+     */
+    @Test
+    public void canRemoveTripHistoryViaAPI() throws Exception {
+        assumeTrue(IS_END_TO_END);
+        ConnectedDataManager.IS_TEST = true;
+
+        // Set back stop. This allows dates after this to trigger an upload.
+        Date twentyDaysInThePast = getStartOfDay(getDateMinusNumberOfDays(new Date(), 20));
+        TripHistoryUpload tripHistoryUpload = new TripHistoryUpload(twentyDaysInThePast);
+        tripHistoryUpload.status = TripHistoryUploadStatus.COMPLETED.getValue();
+        Persistence.tripHistoryUploads.create(tripHistoryUpload);
+
+        // Create OTP user and trip data.
+        otpUser = PersistenceTestUtils.createUser("test@example.com");
+        Date tenDaysInThePast = getStartOfDay(getDateMinusNumberOfDays(new Date(), 10));
+        tripRequest = PersistenceTestUtils.createTripRequest(otpUser.id, tenDaysInThePast);
+        tripSummary = PersistenceTestUtils.createTripSummary(tripRequest.id, tenDaysInThePast);
+
+        // Update 'storeTripHistory' value.
+        otpUser.storeTripHistory = false;
+        mockAuthenticatedRequest(
+            String.format("%s/%s", OTP_USER_PATH, otpUser.id),
+            JsonUtils.toJson(otpUser),
+            otpUser,
+            HttpMethod.PUT
+        );
+        assertEquals(
+            1,
```

Comment:
We expect this because we've only created one TripHistory, right?

Reply:
yeah, I'll add a comment.